lib/theme_check/template.rb

```diff
@@ -3,16 +3,25 @@
 
 module ThemeCheck
   class Template
-    attr_reader :path
+    def initialize(relative_path, storage)
+      @storage = storage
+      @relative_path = relative_path
+    end
 
-    def initialize(path, root)
-      @path = Pathname(path)
-      @root = Pathname(root)
-      @updated = false
+    def path
+      @storage.path(@relative_path)
     end
 
     def relative_path
-      @path.relative_path_from(@root)
+      @relative_pathname ||= Pathname.new(@relative_path)
+    end
+
+    def source
+      @storage.read(@relative_path)
```

Comment:
This probably should be memoized like before. Source will not change, and we read the template source from a few different places.

Reply:
Indeed!